File: Corte 2/Taller 1/Seleccion.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define N 1000

void selectionSort(int arr[], int n, int *comparisons, int *swaps) {
    int i, j, min_idx, temp;
    for (i = 0; i < n - 1; i++) {
        min_idx = i;
        for (j = i + 1; j < n; j++) {
            (*comparisons)++;
            if (arr[j] < arr[min_idx]) {
                min_idx = j;
            }
        }

        // Swap
        if (min_idx != i) {
            temp = arr[i];
            arr[i] = arr[min_idx];
            arr[min_idx] = temp;
            (*swaps)++;
        }
    }
}

int main() {
    int arr[N];
    int comparisons = 0, swaps = 0;

    // Generar números aleatorios de 0 a 9
    srand(time(0));
    for (int i = 0; i < N; i++) {
        arr[i] = rand() % 10; // Números entre 0 y 9
    }

    // Mostrar números generados
    printf("Numeros generados:\n");
    for (int i = 0; i < N; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    // Ordenar usando el algoritmo de selección
    selectionSort(arr, N, &comparisons, &swaps);

    // Mostrar números ordenados
    printf("Numeros ordenados:\n");
    for (int i = 0; i < N; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    // Mostrar resultados
    printf("Comparaciones: %d\n", comparisons);
    printf("Intercambios: %d\n", swaps);
    printf("Estable: No\n");
    printf("Insercion: No\n");

    return 0;
}

File: Corte 2/Taller 1/Algoritmo de Ordenamiento Go/Seleccion.go
package main

import (
    "fmt"
    "math/rand"
    "time"
)

const N = 1000

func selectionSort(arr []int) (comparisons int, swaps int) {
    for i := 0; i < N-1; i++ {
        max_idx := i
        for j := i + 1; j < N; j++ {
            comparisons++
            if arr[j] > arr[max_idx] { // Cambio aquí: arr[j] > arr[max_idx]
                max_idx = j
            }
        }

        // Swap
        if max_idx != i {
            arr[i], arr[max_idx] = arr[max_idx], arr[i]
            swaps++
        }
    }
    return
}

func main() {
    rand.Seed(time.Now().UnixNano())

    // Generar números aleatorios de 0 a 9
    arr := make([]int, N)
    for i := range arr {
        arr[i] = rand.Intn(10) // Números entre 0 y 9
    }

    // Mostrar números generados
    fmt.Println("Numeros generados:")
    for _, num := range arr {
        fmt.Printf("%d ", num)
    }
    fmt.Println()

    // Ordenar usando el algoritmo de selección
    comparisons, swaps := selectionSort(arr)

    // Mostrar números ordenados
    fmt.Println("Numeros ordenados:")
    for _, num := range arr {
        fmt.Printf("%d ", num)
    }
    fmt.Println()

    fmt.Printf("Comparaciones: %d\n", comparisons)
    fmt.Printf("Intercambios: %d\n", swaps)
    fmt.Println("Estable: No")
    fmt.Println("Insercion: No")
}